state/sqlite/sqlite_dbaccess.go

```diff
@@ -246,50 +247,92 @@ func (a *sqliteDBAccess) Get(parentCtx context.Context, req *state.GetRequest) (
 	if req.Key == "" {
 		return nil, errors.New("missing key in get operation")
 	}
-	var (
-		value    []byte
-		isBinary bool
-		etag     string
-	)
 
-	// Sprintf is required for table name because sql.DB does not substitute parameters for table names
-	//nolint:gosec
-	stmt := fmt.Sprintf(
-		`SELECT value, is_binary, etag FROM %s
+	// Concatenation is required for table name because sql.DB does not substitute parameters for table names
+	stmt := `SELECT key, value, is_binary, etag FROM ` + a.metadata.TableName + `
 		WHERE
 			key = ?
-			AND (expiration_time IS NULL OR expiration_time > CURRENT_TIMESTAMP)`,
-		a.metadata.TableName)
+			AND (expiration_time IS NULL OR expiration_time > CURRENT_TIMESTAMP)`
 	ctx, cancel := context.WithTimeout(parentCtx, a.metadata.timeout)
-	err := a.db.QueryRowContext(ctx, stmt, req.Key).
-		Scan(&value, &isBinary, &etag)
+	row := a.db.QueryRowContext(ctx, stmt, req.Key)
 	cancel()
+	_, value, etag, err := readRow(row)
 	if err != nil {
 		if errors.Is(err, sql.ErrNoRows) {
 			return &state.GetResponse{}, nil
 		}
 		return nil, err
 	}
 
+	return &state.GetResponse{
+		Data:     value,
+		ETag:     &etag,
+		Metadata: req.Metadata,
+	}, nil
+}
+
+func (a *sqliteDBAccess) BulkGet(parentCtx context.Context, req []state.GetRequest) (bool, []state.BulkGetResponse, error) {
+	if len(req) == 0 {
+		return true, []state.BulkGetResponse{}, nil
+	}
+
+	// SQLite doesn't support passing an array for an IN clause, so we need to build a custom query
+	inClause := strings.Repeat("?,", len(req))
+	inClause = inClause[:(len(inClause) - 1)]
```

Comment:
Maybe do this instead
```suggestion
	inClause := strings.Join(strings.Repeat("?", len(req)), ",")
```

Reply:
I'm afraid this won't work as `strings.Repeat` returns a string, while `strings.Join` wants a slice. We could of course allocate a slice, but that would require more allocations on the heap and doesn't seem as efficient as the code above (which I reckon isn't the prettiest, but should be fairly efficient and this is a hot path).